src/operator/tensor/init_op.h

```diff
@@ -11,20 +11,80 @@
 #include <mxnet/op_attr_types.h>
 #include <dmlc/parameter.h>
 #include <vector>
+#include <string>
+#include <limits>
 #include "../elemwise_op_common.h"
 
 namespace mxnet {
 namespace op {
 
 struct InitOpParam : public dmlc::Parameter<InitOpParam> {
   TShape shape;
+  std::string ctx;
+  int dtype;
   DMLC_DECLARE_PARAMETER(InitOpParam) {
     DMLC_DECLARE_FIELD(shape)
     .set_default(TShape())
     .describe("The shape of the output");
+    DMLC_DECLARE_FIELD(dtype)
+    .set_default(mshadow::kFloat32)
+    .describe("DType of the output");
+    DMLC_DECLARE_FIELD(ctx)
```

Comment:
After the change to strict checking, its probably better to use `__ctx__` to avoid having to add these in operators. @tqchen What do you think